### PR TITLE
meta: merge main*, add TODOs and implement fixes for DataTypes PR

### PR DESCRIPTION
**Pull request by WikiRik**
## Pull Request Checklist

- [ ] Have you added new tests to prevent regressions?
- [ ] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [ ] Did you update the typescript typings accordingly (if applicable)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change
This is the superseded PR from #15209. Same idea, just not with the merge conflicts of the hooks migration fixed.

It was getting harder to go through the unresolved conversations in the DataTypes PR so I collected them all here in TODO!s (or I implemented a fix directly myself). Once we've solved those I think we're good to merge the DataTypes PR.

Most of the changes are due to merging `main` in this. To properly review the PR I would recommend filtering for the changes in the commit after the merge.

One commit of main is not merged in here and that's the Hooks migration. It caused some issues (see #15209 )

@ephys let me know if this helps you out, if not we can always close this again and use https://github.com/sequelize/sequelize/commit/e617b5c2e11d8dabe174c52aa8d1cd7a057f5d9e to check what still needs to be done

## Todos

- [ ] Resolve TODO!s
- [ ] Merge main completely in here

### CODE REVIEW

**Review by fzn0x — APPROVED** [review]

**Comment by fzn0x** [comment]
Hi @WikiRik can you dismiss my review? As it-is a draft and I mistakenly approved it.

**Comment by fzn0x** [comment]
Hi @WikiRik can you dismiss my approval? Since it is a draft and I mistakenly approved it.

**Comment by WikiRik** [comment]
Closing this, main has been merged in the original PR and final conversations there have been unresolved making this PR redundant

File: src/dialects/abstract/data-types.ts [hunk 1]
```diff
@@ -1354,6 +1354,7 @@ export class DATEONLY extends AbstractDataType<AcceptedDate> {
     }
 
     if (value) {
+      // TODO!: add a test case for #10982
```
Comment (ephys) [inline src/dialects/abstract/data-types.ts:1357]: Here

File: src/dialects/abstract/query-generator.js [hunk 2]
```diff
@@ -2919,6 +2919,7 @@ Only named replacements (:name) are allowed in literal() because we cannot guara
       }, escapeOptions), comparator, options.prefix);
     }
 
+    // TODO!: determine whether we're missing something or that this coment can be removed; https://github.com/sequelize/sequelize/pull/14505/commits/56828b202a6f2ea11a185d1ec9d40a776cac7257#diff-54c14e4744281a47d62196b5892c4004fec6e2f3c4d90f444aa3bd0a880f2209
```
Comment (ephys) [inline src/dialects/abstract/query-generator.js:2922]: Here

File: test/integration/data-types/data-types.test.ts [hunk 3]
```diff
@@ -215,7 +215,7 @@ describe('DataTypes', () => {
       expect(pick(user.get(), ['tinyText', 'mediumText', 'longText'])).to.deep.eq(data);
     });
 
-    // TODO: once we have centralized logging, check a warning message has been emitted when length is not supported:
+    // TODO!: once we have centralized logging, check a warning message has been emitted when length is not supported:
```
Comment (ephys) [inline test/integration/data-types/data-types.test.ts:218]: Here

File: test/unit/model/define.test.js [hunk 4]
```diff
@@ -195,7 +195,7 @@ describe(Support.getTestDialectTeaser('Model'), () => {
       }).to.throw('Invalid definition for "part.name", "notNull" validator is only allowed with "allowNull:false"');
     });
 
-    // TODO: once we have centralized logging, enable this again
+    // TODO!: once we have centralized logging, enable this again
```
Comment (ephys) [inline test/unit/model/define.test.js:198]: Here

File: test/unit/sql/insert.test.js [hunk 5]
```diff
@@ -93,6 +93,7 @@ describe(Support.getTestDialectTeaser('SQL'), () => {
           bind: {
             ibmi: { sequelize_1: '2015-01-20 00:00:00.000' },
             sqlite: { sequelize_1: '2015-01-20 00:00:00.000 +00:00' },
+            // TODO!: for this test and the test below check why the result is different
```
Comment (ephys) [inline test/unit/sql/insert.test.js:96]: And here

**Review by ephys — COMMENTED** [review]
There are a few TODOs that aren't handled in the target PR. This is just to point them out

File: test/integration/data-types/data-types.test.ts (hunk 3 above)
Comment (WikiRik) [reply test/integration/data-types/data-types.test.ts:218]: I've regarded this a fine one for a future PR, the others have been unresolved in the main PR

File: test/unit/sql/insert.test.js (hunk 5 above)
Comment (WikiRik) [reply test/unit/sql/insert.test.js:96]: This one has been resolved in the main PR by https://github.com/sequelize/sequelize/pull/14505/commits/a71035e67c72ff62a70d46fe192f8ccd0157319d

File: src/dialects/abstract/data-types.ts (hunk 1 above)
Comment (WikiRik) [reply src/dialects/abstract/data-types.ts:1357]: That has been added in https://github.com/sequelize/sequelize/pull/14505/commits/77e70f1d4b6b9c59a70dc8983fecb30c970fc3bd

File: src/dialects/abstract/query-generator.js (hunk 2 above)
Comment (WikiRik) [reply src/dialects/abstract/query-generator.js:2922]: Fixed in https://github.com/sequelize/sequelize/pull/14505/commits/2a9efd24385161503fd880f5cac2d3e2c3f0e93a

File: test/unit/model/define.test.js (hunk 4 above)
Comment (WikiRik) [reply test/unit/model/define.test.js:198]: Fixed in https://github.com/sequelize/sequelize/pull/14505/commits/0f62d00fec8df539c1ca4086fc2abf917019b28c

**Pull request closed without merging**